pulsar/consumer_multitopic.go

```diff
@@ -167,6 +167,47 @@ func (c *multiTopicConsumer) AckID(msgID MessageID) error {
 	return mid.consumer.AckID(msgID)
 }
 
+func (c *multiTopicConsumer) AckIDList(msgIDs []MessageID) error {
+	return ackIDListFromMultiTopics(c.log, msgIDs, func(msgID MessageID) (acker, error) {
+		if !checkMessageIDType(msgID) {
+			return nil, errors.New("invalid message id type %T")
+		}
+		if mid := toTrackingMessageID(msgID); mid != nil && mid.consumer != nil {
+			return mid.consumer, nil
+		}
+		return nil, errors.New("consumer is nil")
+	})
+}
+
+func ackIDListFromMultiTopics(log log.Logger, msgIDs []MessageID, findConsumer func(MessageID) (acker, error)) error {
+	consumerToMsgIDs := make(map[acker][]MessageID)
+	for _, msgID := range msgIDs {
+		if consumer, err := findConsumer(msgID); err == nil {
+			consumerToMsgIDs[consumer] = append(consumerToMsgIDs[consumer], msgID)
+		} else {
+			log.Warnf("Can not find consumer for %v", msgID)
+		}
+	}
+
+	ackError := make(map[string]*TopicAckError)
+	for consumer, ids := range consumerToMsgIDs {
+		if err := consumer.AckIDList(ids); err != nil {
+			ackError[consumer.Topic()] = &TopicAckError{
+				Err:    err,
+				MsgIDs: ids,
+			}
+		}
+	}
+	if len(ackError) == 0 {
+		return nil
+	} else if len(ackError) == 1 {
+		for _, topicAckError := range ackError {
```

Comment:
As I commented on the API, maybe we could keep it simple and consistently return `AckError`.

Reply:
+1.

Reply:
I simplified the `AckError` with a `map[MessageID]error` now. But I don't include invalid message IDs in this map. PTAL again @shibd @RobertIndie